File: Leetcode/leetcode-0080-RemoveDuplicatesFromSortedArrayII-[array]-[medium]/Solution.two-pointers.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        curVal = 'unknown'
        cnt = 0
        i = 0
        j = 0
        while (i < len(nums)):
            if (nums[i] != curVal):
                curVal = nums[i]
                cnt = 1
                nums[j] = nums[i]
                j = j + 1
            elif (nums[i] == curVal and cnt < 2):
                cnt = cnt + 1
                nums[j] = nums[i]
                j = j + 1

            
            i = i + 1
        
        return j

File: Leetcode/leetcode-0080-RemoveDuplicatesFromSortedArrayII-[array]-[medium]/Solution.two-pointers.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int curVal = Integer.MIN_VALUE;
        int cnt = 0;
        int i = 0, j = 0;
        while (i < nums.length) {
            if (nums[i] != curVal) {
                curVal = nums[i];
                cnt = 1;
                nums[j] = nums[i];
                j++;
            } else if (nums[i] == curVal && cnt < 2) {
                cnt++;
                nums[j] = nums[i];
                j++;
            } 
            
            i++;
        }

        return j;
    }
}